ntoskrnl/ps/query.c

```diff
@@ -175,36 +179,50 @@ NtQueryInformationProcess(
             /* Indicate success */
             Status = STATUS_SUCCESS;
 
-            _SEH2_TRY
+            RtlZeroMemory(&QuotaLimits, sizeof(QuotaLimits));
+
+            /* Get max/min working set sizes */
+            QuotaLimits.MaximumWorkingSetSize =
+                Process->Vm.MaximumWorkingSetSize << PAGE_SHIFT;
+            QuotaLimits.MinimumWorkingSetSize =
+                Process->Vm.MinimumWorkingSetSize << PAGE_SHIFT;
+
+            /* Get default time limits */
+            QuotaLimits.TimeLimit.LowPart = MAXULONG;
+            QuotaLimits.TimeLimit.HighPart = MAXULONG;
+
+            /* Is quota block a default one? */
+            if (Process->QuotaBlock == &PspDefaultQuotaBlock)
             {
-                /* Set max/min working set sizes */
-                QuotaLimits->MaximumWorkingSetSize =
-                        Process->Vm.MaximumWorkingSetSize << PAGE_SHIFT;
-                QuotaLimits->MinimumWorkingSetSize =
-                        Process->Vm.MinimumWorkingSetSize << PAGE_SHIFT;
+                /* Get default pools and pagefile limits */
+                QuotaLimits.PagedPoolLimit = (SIZE_T)-1;
+                QuotaLimits.NonPagedPoolLimit = (SIZE_T)-1;
+                QuotaLimits.PagefileLimit = (SIZE_T)-1;
```

Comment:
Isn't it possible to just feed these values by copying them from the `PspDefaultQuotaBlock` ?

Reply:
Not sure, but I think the original code is OK.